channeldb/control_tower.go

```diff
@@ -92,6 +116,9 @@ func (p *paymentControl) InitPayment(paymentHash lntypes.Hash,
 	}
 	infoBytes := b.Bytes()
 
+	p.hashMutex.Lock(paymentHash)
```

Comment:
should have a comment

Reply:
Added

Reply:
Also removed locks that were present at init and registerattempt. I don't think they were necessary.